ddtrace/tracer.py

```diff
@@ -694,6 +694,8 @@ def _start_span(
                 # We must always mark the span as sampled so it is forwarded to the agent
                 span.sampled = True
 
+        span.set_tag_str("language", "python")
```

Comment:
We might want to performance test this change, since this will add encoding overhead to all spans.


Given that there are different ways for us to get this data to the trace agent/intake we might want to defer the implementation detail to the encoder/writer.

By that I mean instead of setting the language tag on every span, add it as a default tag on each `Context` object.

Then we can either:

- Ignore the tag if we have trace agent >= 7.33 since we get the value from setting the header
- Have the encoder add this to every span, similar to what we do for `_dd.origin` tag.


cc @Kyle-Verhoog who I am guessing will have some good thoughts/opinions here.

Reply:
I'm not sure how the first option works with setting the header, could you explain? Thanks!

Reply:
The trace agent in version 7.33 will also pass along the value of `Datadog-Meta-Lang` header to the backend in a new payload format. The backend will then ensure all spans in that payload have the language tag present and set to value of `Datadog-Meta-Lang`.

This is the commit in the agent which adds it: https://github.com/DataDog/datadog-agent/commit/4b005758cdf6b5a63c9f3957ad8d82c17b9c558a

This is the code that adds the header value to the payloads: https://github.com/DataDog/datadog-agent/blob/4b005758cdf6b5a63c9f3957ad8d82c17b9c558a/pkg/trace/api/api.go#L401-L445